Refit.Tests/GitHubApi.cs

```diff
@@ -76,5 +76,9 @@ public interface IGitHubApi
 
         [Get("/give-me-some-404-action")]
         Task NothingToSeeHere();
+
+        [NullOn404]
+        [Get("/give-me-some-404-action")]
+        Task<object> ReturnsNullOn404();
```

Comment:
Add test for value types too

Reply:
The null filter can't work with value types. Do you mean adding a test to make sure value types won't go through the filter and will throw the regular exception? Or do you mean changing the implementation to a `DefaultOn404`?

Reply:
It should still work with `Task<int> ReturnsNullOn404()` probably throwing an exception or something like that.

Reply:
I think Default makes more sense than Null as it covers all the bases. That said, I'm not sure if all of these attributes belong in Refit at all. The base implementation, perhaps, but I'm not sure what's special about 404 to warrant it having an implementation here when all of the other responses aren't touched.